maven/README.md

```diff
@@ -157,49 +157,71 @@ spec:
         cat > /.m2/settings.xml <<EOF
         <settings>
           <mirrors>
-            <mirror>
-              <id>mirror.default</id>
-              <name>mirror.default</name>
-              <url>$(inputs.params.MAVEN_MIRROR_URL)</url>
-              <mirrorOf>*</mirrorOf>
-            </mirror>
+            <!-- ### configured mirrors ### -->
           </mirrors>
           <proxies>
-            <proxy>
-              <id>proxy.default</id>
-              <active>true</active>
-              <protocol>$(inputs.params.PROXY_PROTOCOL)</protocol>
-              <username>$(inputs.params.PROXY_USER)</username>
-              <password>$(inputs.params.PROXY_PASSWORD)</password>
-              <host>$(inputs.params.PROXY_HOST)</host>
-              <port>$(inputs.params.PROXY_PORT)</port>
-              <nonProxyHosts>$(inputs.params.NON_PROXY_HOST)</nonProxyHosts>
-            </proxy>
+            <!-- ### configured http proxy ### -->
           </proxies>
         </settings>
         EOF
 
+        if [ -n "$(inputs.params.PROXY_HOST)" -a -n "$(inputs.params.PROXY_PORT)" ]; then
+          xml="<proxy>\
+            <id>genproxy</id>\
+            <active>true</active>\
+            <protocol>$(inputs.params.PROXY_PROTOCOL)</protocol>\
+            <host>$(inputs.params.PROXY_HOST)</host>\
+            <port>$(inputs.params.PROXY_PORT)</port>"
+          if [ -n "$(inputs.params.PROXY_USER)" -a -n "$(inputs.params.PROXY_PASSWORD)" ]; then
+            xml="$xml\
+                <username>$(inputs.params.PROXY_USER)</username>\
+                <password>$(inputs.params.PROXY_PASSWORD)</password>"
+          fi
+          if [ -n "$(inputs.params.HTTP_PROXY_NONPROXYHOSTS)" ]; then
+            xml="$xml\
+                <nonProxyHosts>$(inputs.params.HTTP_PROXY_NONPROXYHOSTS)</nonProxyHosts>"
+          fi
+          xml="$xml\
+              </proxy>"
+          sed -i "s|<!-- ### configured http proxy ### -->|$xml|" /.m2/settings.xml
+        fi
+
+        if [ -n "$(inputs.params.MAVEN_MIRROR_URL)" ]; then
+          xml="    <mirror>\
+            <id>mirror.default</id>\
+            <url>$(inputs.params.MAVEN_MIRROR_URL)</url>\
+            <mirrorOf>external:*</mirrorOf>\
+          </mirror>"
+          sed -i "s|<!-- ### configured mirrors ### -->|$xml|" /.m2/settings.xml
+        fi
+
         [[ -f /.m2/settings.xml ]] && cat /.m2/settings.xml
         [[ -f /.m2/settings.xml ]] || echo skipping settings
+
       volumeMounts:
         - name: m2-repository
           mountPath: /.m2
-        - name: maven-configmap
-          mountPath: /.m2/settings.xml
-          subPath: settings.xml
+
     - name: mvn-goals
       image: gcr.io/cloud-builders/mvn
-      command:
-        - /usr/bin/mvn
       args:
         - "$(inputs.params.GOALS)"
+      script: |
+        #!/bin/bash
+
+        if [[ -f "/.m2/settings.xml" ]]
+        then
+          mvn ${@} --settings /.m2/settings.xml
```

Comment:
maybe put the `${@}` after `--settings` ? Not sure if this works, but I would expect if multiple settings are given then the last one "wins", which I would expect to be the one provided the command line.